model registry: pin tag and standalone instructions (#3740)
kubeflow/website: content/en/docs/components/model-registry/installation.md

<a id="model-registry-install"></a>

> ## Installing Model Registry (as part of Kubeflow)

Reviewer: @tarilabs Perhaps we could do two H3 subsections, so this becomes

```
## Installing Model Registry

Kubeflow Model registry may be installed as part of a Kubeflow Platform, or as a standalone component.
The best option for you will depend on your specific requirements.

### Kubeflow Platform

Kubeflow Model Registry is available in Kubeflow Platform 1.9+, see [Installing Kubeflow](/docs/started/installing-kubeflow/) to learn more about deploying the Kubeflow Platform.

If you have deployed the raw manifests, you may follow [these instructions]((https://github.com/kubeflow/manifests/tree/master/apps/model-registry/upstream#readme), please raise any feedback on [`kubeflow/manifests#2631`](https://github.com/kubeflow/manifests/issues/2631).

### Standalone

.... Guide ....

Author: implemented with https://github.com/kubeflow/website/pull/3740/commits/1542df1079e9cbc13959a1345a8045940b84cd42

thank you; resolving.

## Installing Model Registry

Kubeflow Model registry may be installed as part of a Kubeflow Platform, or as a standalone component.
The best option for you will depend on your specific requirements.

### Installing on Kubeflow Platform

Kubeflow Model Registry is available as an opt-in alpha component in Kubeflow Platform 1.9+, see [Installing Kubeflow](/docs/started/installing-kubeflow/) to learn more about deploying the Kubeflow Platform.

> If you have deployed the raw manifests, you may follow [these instructions](https://github.com/kubeflow/manifests/tree/master/apps/model-registry/upstream#readme) to deploy Model Registry; please raise any feedback on [`kubeflow/manifests#2631`](https://github.com/kubeflow/manifests/issues/2631).

Reviewer 2: Let's call them Kubeflow Manifests.
```suggestion
If you have deployed the Kubeflow manifests, you may follow [these instructions](https://github.com/kubeflow/manifests/tree/master/apps/model-registry/upstream#readme) to deploy Model Registry; please raise any feedback on [`kubeflow/manifests#2631`](https://github.com/kubeflow/manifests/issues/2631).
```

> If you have deployed the raw manifests, you may follow [these instructions](https://github.com/kubeflow/manifests/tree/master/apps/model-registry/upstream#readme) to deploy Model Registry; please raise any feedback on [`kubeflow/manifests#2631`](https://github.com/kubeflow/manifests/issues/2631).

Reviewer 2: Do you want to create an issue in the model-registry repo, not in the manifests ?
So it will be easier for your to track and triage it ?

Author: oh I see, I wanted them to refer to the original issue, but you are probably right they might go off and create new one (but in that case they might be using the "wrong" repo). 
amended with 276c49eb9e954d960e2c256aaf019f5827456ade

### Standalone installation

If you want to install Model Registry separately from Kubeflow, or to get a later version

> of Model Registry, you can the Model Registry manifests directly from the [Model Registry repository](https://github.com/kubeflow/model-registry).

Reviewer 2: ```suggestion
of Model Registry, you can install the Model Registry manifests directly from the [Model Registry repository](https://github.com/kubeflow/model-registry).
```

The manifests by defaut deploy the Model Registry in the `kubeflow` namespace;
you must ensure the `kubeflow` namespace is available (e.g.: `kubectl create namespace kubeflow`)
or modify the [kustomization file](https://github.com/kubeflow/model-registry/blob/v0.2.0-alpha/manifests/kustomize/overlays/db/kustomization.yaml#L3) accordingly to your setup.

Remember to substitute the relevant release version number, by modifying the actual `ref=` to your desired version number.
For example, modify `ref=v0.2.0-alpha` to `ref=v0.3.2` if you want to deploy version 0.3.2 instead.

```shell
kubectl apply -k "https://github.com/kubeflow/model-registry/manifests/kustomize/overlays/db?ref=v0.2.0-alpha"
```

You may also apply the Istio-related manifests if your cluster installation provides an Istio-mesh; these are also required when using Model Registry as part of Kubeflow.

```shell

> kubectl apply -k "https://github.com/kubeflow/model-registry/manifests/kustomize/options/istio?ref=v0.2.0-alpha"

Reviewer 2: Can users take the kustomize manifests from `main` branch to install the latest changes ?
Or you push images to the DockerHub only when you do release tag ?

Author: done with https://github.com/kubeflow/website/pull/3740/commits/b75772fd609ddf28679c1708ad519d2cd192dcab

Author: Resolving this thread based on feedback from : https://github.com/kubeflow/website/pull/3740#pullrequestreview-2093485477

## Installing Model Registry

Kubeflow Model registry may be installed as part of a Kubeflow Platform, or as a standalone component.
The best option for you will depend on your specific requirements.

### Installing on Kubeflow Platform

Kubeflow Model Registry is available as an opt-in alpha component in Kubeflow Platform 1.9+, see [Installing Kubeflow](/docs/started/installing-kubeflow/) to learn more about deploying the Kubeflow Platform.

> If you have deployed the Kubeflow manifests, you may follow [these instructions](https://github.com/kubeflow/manifests/tree/master/apps/model-registry/upstream#readme) to deploy Model Registry; please raise any feedback on [`kubeflow/model-registry`](https://github.com/kubeflow/model-registry/issues/new/choose).

Reviewer: Usually, it's best to not link to the "new issue" page, so it encourages people to actually search existing issues first.

We should link to `https://github.com/kubeflow/model-registry/issues/` instead.

Author: done with 593e2335940bb6eacc5bbcffec34df12c2595730 thanks

### Standalone installation

If you want to install Model Registry separately from Kubeflow, or to get a later version
of Model Registry, you can install the Model Registry manifests directly from the [Model Registry repository](https://github.com/kubeflow/model-registry).

> The manifests by defaut deploy the Model Registry in the `kubeflow` namespace;

Reviewer: The word "default "was misspelled, but also its cleaner to phrase it like this:

```suggestion
By default, the manifests deploy the Model Registry in the `kubeflow` namespace;
```

> you must ensure the `kubeflow` namespace is available (e.g.: `kubectl create namespace kubeflow`)

Reviewer: ```suggestion
you must ensure the `kubeflow` namespace is available (for example: `kubectl create namespace kubeflow`)
```

Author: out of curiosity, in https://github.com/kubeflow/website/pull/3740#discussion_r1628170787 you suggested yourself using `e.g.:`
while this recommendation is replacing it with `for example:` 

... any particular guideline you are applying? 🤔

> or modify the [kustomization file](https://github.com/kubeflow/model-registry/blob/v0.2.0-alpha/manifests/kustomize/overlays/db/kustomization.yaml#L3) accordingly to your setup.

Reviewer: ```suggestion
or modify [the kustomization file](https://github.com/kubeflow/model-registry/blob/v0.2.0-alpha/manifests/kustomize/overlays/db/kustomization.yaml#L3) to your desired namespace.
```

> 

Reviewer: We should add a section here that explains how to get the list of currently tagged releases.

```suggestion

See the list of available versions on the [GitHub releases](https://github.com/kubeflow/model-registry/releases) of the `kubeflow/model-registry` repository. To install a specific release of the Model Registry, modify the following commands with the desired `ref=<GIT_TAG>`.

```

Author: great suggestion, thanks!

> Run the following command to install the latest alpha release version of Model Registry:

Reviewer: It's probably best if we don't imply that this is the "latest" because it will become out of date, let's just say a specific version:

```suggestion
Run the following command to install the `v0.2.0-alpha` release of Model Registry:
```

```shell
kubectl apply -k "https://github.com/kubeflow/model-registry/manifests/kustomize/overlays/db?ref=v0.2.0-alpha"
```

> You may also apply the Istio-related manifests if your cluster installation provides an Istio-mesh; these are also required when using Model Registry as part of Kubeflow.

Reviewer: I think this can be rephrased to make it clear who needs to run this command:

```suggestion
If your Kubernetes cluster uses Istio, you MUST apply the Istio-compatibility manifests (e.g. when using a full Kubeflow Platform). However, these are NOT required for non-Istio clusters.
```

```shell
kubectl apply -k "https://github.com/kubeflow/model-registry/manifests/kustomize/options/istio?ref=v0.2.0-alpha"
```

> If you want instead to install the latest changes or another release of Model Registry, modify the above commands with the desired `ref=`.
> For example, modify `ref=v0.2.0-alpha` to `ref=main` if you want to deploy the latest changes directly from the main branch instead.

Reviewer: If we make the previous suggestions, this information is already stated above.

Also, I would strongly recommend against telling people to apply `main`, as we want to be able to make breaking changes in main without having people break their clusters.
```suggestion
```
